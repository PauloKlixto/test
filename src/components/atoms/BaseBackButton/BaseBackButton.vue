<template>
  <section>
    <component
      :is="tag"
      class="BaseBackButton"
      :class="BaseBackButtonClasses"
      @click="click"
    >
      {{ label }}
    </component>
  </section>
</template>

<script>
export default {
  name: 'BaseBackButton',

  props: {
    tag: {
      type: String,
      default: 'a',
      validator: val => ['button', 'a'].includes(val)
    },
    label: {
      type: String,
      default: ''
    },
    isInvertedTheme: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    BaseBackButtonClasses() {
      return {
        'BaseBackButton--invert': this.isInvertedTheme
      }
    }
  },

  methods: {
    click(e) {
      this.$emit('click', e)
    }
  }
}
</script>

<style lang="scss" scoped>
.BaseBackButton {
  font-size: 13px;
  padding-left: 20px;
  display: block;
  position: relative;
  cursor: pointer;

  &:before {
    content: '';
    width: 8px;
    height: 8px;
    border-top: 2px solid $color-white;
    border-left: 2px solid $color-white;
    position: absolute;
    left: 0;
    transform: rotateZ(-45deg) translateY(5px);
  }

  &--invert {
    color: $color-white;
  }
}
</style>
