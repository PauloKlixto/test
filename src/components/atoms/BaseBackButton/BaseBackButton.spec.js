/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import BaseBackButton from './BaseBackButton'

describe('BaseBackButton', () => {
  const WRAPPER = shallowMount(BaseBackButton)

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Emitar funcao click', () => {
    const MOCK = { test: 'test' }
    WRAPPER.vm.click(MOCK)
    expect(WRAPPER.emitted('click')).toEqual([[MOCK]])
  })

  test('Componente nÃ£o alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})
