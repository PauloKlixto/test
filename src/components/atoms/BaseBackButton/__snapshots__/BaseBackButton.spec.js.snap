// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBackButton Componente n√£o alterado 1`] = `
"<section><a class=\\"BaseBackButton\\">

  </a></section>"
`;
